src/actions/ActionsMakeshift.ts

```diff
@@ -0,0 +1,86 @@
+import { Target } from "../typings/target.types";
+import sleep from "../util/sleep";
+import CommandAction from "./CommandAction";
+
+abstract class MakeshiftAction extends CommandAction {
+  abstract command: string;
+  restoreSelection?: boolean;
+  commandArg?: object;
+  ensureSingleTarget?: boolean;
+  postCommandSleepMs?: number;
+
+  async run(targets: [Target[]]) {
+    return super.run(targets, {
+      command: this.command,
+      commandArgs: this.commandArg ? [this.commandArg] : [],
+      ensureSingleTarget: this.ensureSingleTarget ?? false,
+      restoreSelection: this.restoreSelection ?? true,
+    });
+    if (this.postCommandSleepMs) {
+      await sleep(this.postCommandSleepMs);
+    }
+  }
+}
+
+export class RevealDefinition extends MakeshiftAction {
+  command = "editor.action.revealDefinition";
+  ensureSingleTarget = true;
+  restoreSelection = false;
+}
+
+export class RevealTypeDefinition extends MakeshiftAction {
+  command = "editor.action.goToTypeDefinition";
+  ensureSingleTarget = true;
+  restoreSelection = false;
+}
+
+export class ShowHover extends MakeshiftAction {
+  command = "editor.action.showHover";
+  ensureSingleTarget = true;
+  restoreSelection = false;
+}
+
+export class ShowDebugHover extends MakeshiftAction {
+  command = "editor.debug.action.showDebugHover";
+  ensureSingleTarget = true;
+  restoreSelection = false;
+}
+
+export class ShowQuickFix extends MakeshiftAction {
+  command = "editor.action.quickFix";
+  ensureSingleTarget = true;
+  postCommandSleepMs = 100;
+}
+
+export class ShowReferences extends MakeshiftAction {
+  command = "references-view.find";
+  ensureSingleTarget = true;
+}
+
+export class Rename extends MakeshiftAction {
+  command = "editor.action.rename";
+  ensureSingleTarget = true;
+  postCommandSleepMs = 200;
```

Comment:
This sleep doesn't matter since talon is not waiting for this command

Reply:
How come it is here then?

Reply:
Because I wanted to have a discussion about it. This sleep is necessary for mac os afaik. So we have to put it somewhere.